fido2/attestation.py

```diff
@@ -242,3 +245,38 @@ def verify(self, statement, auth_data, client_data_hash):
             pub_key.verify(auth_data + client_data_hash, statement["sig"])
         except _InvalidSignature:
             raise InvalidSignature()
+
+
+class TpmAttestation(Attestation):
+    FORMAT = "tpm"
+
+    def verify(self, statement, auth_data, client_data_hash):
+        if "ecdaaKeyId" in statement:
+            raise NotImplementedError("ECDAA not implemented")
+        alg = statement["alg"]
+        x5c = statement.get("x5c")
+        cert_info = statement["certInfo"]
+        if x5c:
+            cert = x509.load_der_x509_certificate(x5c[0], default_backend())
+            _validate_attestation_certificate(
+                cert, auth_data.credential_data.aaguid, check_subject=False
+            )
+
+            pub_key = CoseKey.for_alg(alg).from_cryptography_key(cert.public_key())
+        else:
+            pub_key = CoseKey.parse(auth_data.credential_data.public_key)
+            if pub_key.ALGORITHM != alg:
+                raise InvalidData("Wrong algorithm of public key!")
+        try:
+            att_to_be_signed = auth_data + client_data_hash
+            digest = hashes.Hash(pub_key.HASH_ALG, backend=default_backend())
```

Comment:
Every cose key now exposes the HASH_ALG, because spec defines:
>  setting the extraData parameter to the digest of attToBeSigned using the hash algorithm corresponding to the "alg" signature algorithm. (For the "RS256" algorithm, this would be a SHA-256 digest.)

The only key not to is EdDSA but I'm not too sure we should expect a tpm to use EdDSA keys

Reply:
I'd prefer CoseKey.HASH_ALG not being an instance of a Cryptography class, but I'm also not completely sure about what I think we should do instead. Using a string identifier such as "sha256" could be an option, but going from that to a Cryptography HashAlgorithm instance then becomes messy. I'm a bit torn.

Reply:
let me know what you want to do. I'd be happy to change that.

Reply:
how about prefixing with `_`, to just show this is private api?

Reply:
Yeah, that sounds good, let's go with that.